ompi/mca/common/configure.m4

```diff
@@ -0,0 +1,20 @@
+# -*- shell-script -*-
+#
+# Copyright (c) 2006-2007 Los Alamos National Security, LLC.
+#                         All rights reserved.
+# Copyright (c) 2012      Cisco Systems, Inc.  All rights reserved.
+# Copyright (c) 2016      Research Organization for Information Science
+#                         and Technology (RIST). All rights reserved.
+# $COPYRIGHT$
+#
+# Additional copyrights may follow
+#
+# $HEADER$
+#
+
+# MCA_ompi_common_CONFIG(project_name, framework_name)
+# -------------------------------------------
+AC_DEFUN([MCA_ompi_common_CONFIG],
+[
+    MCA_CONFIGURE_FRAMEWORK([$1], [$2], [$define_mpi_io])
```

Comment:
...actually, after my first "oh, that's cool!" reaction, I don't think that this is *quite* right yet.

This solution will disable *all* common components when `--disable-mpi-io` is used, even if those common components have nothing to do with MPI IO.

Granted, right now, there's only `ompi/mca/common/ompio`, but there could well be other common components that are unrelated to MPI IO (we have a bunch in `opal/mca/common`, for example).

Can you adjust?